visualization/app/codeCharta/ui/ribbonBar/colorSettingsPanel/colorSettingsPanel.component.ts

```diff
@@ -19,14 +24,44 @@ import { calculateInitialColorRange } from "../../../state/store/dynamicSettings
 	encapsulation: ViewEncapsulation.None
 })
 export class ColorSettingsPanelComponent {
-	isDeltaState$ = this.store.select(isDeltaStateSelector)
 	colorMode$ = this.store.select(colorModeSelector)
 	colorLabels$ = this.store.select(colorLabelsSelector)
+	colorMetric$ = this.store.select(colorMetricSelector)
+	isDeltaState$ = this.store.select(isDeltaStateSelector)
+	sliderValues$ = this.store.select(metricColorRangeValuesSelector)
+	sliderColors$ = this.store.select(metricColorRangeColorsSelector)
 	isColorRangeInverted = false
 	areDeltaColorsInverted = false
 
+	private newLeftValue: null | number = null
+	private newRightValue: null | number = null
+
 	constructor(private store: Store<CcState>, private State: State<CcState>) {}
 
+	handleValueChange: HandleValueChange = ({ newLeftValue, newRightValue }) => {
+		if (newLeftValue !== undefined) {
+			this.newLeftValue = newLeftValue
+		}
+		if (newRightValue !== undefined) {
+			this.newRightValue = newRightValue
+		}
```

Comment:
Should work as well:
```suggestion
               this.newLeftValue = newLeftValue ?? this.newLeftValue;
               this.newRightValue = newRightValue ?? this.newRightValue;
```